### DISCUSSION TITLE
Why is there a remote stop sensor capability

### DISCUSSION THREAD

**Discussion by Navid200** [Navid200/xDrip#16] [Development] [answered]
This is a section of the GcmListenerSvc.java file July 16, 2020.

![Screenshot 2021-05-15 125009](https://user-images.githubusercontent.com/51497406/118371861-7943f100-b57c-11eb-998a-bfb6a98da7f5.png)

What is this for?  
Does this mean someone who is following can stop sensor?

**Comment by tzachi-dar** [comment, marked as answer]
For young kids, the parent who is many times remote should stop the sensor, and start another.

**Reply by Navid200** [reply]
How does a parent stop a sensor and place a new one remotely?

**Reply by Navid200** [reply]
I know how they stop it.  But, how would they place a new sensor on without the child or someone else doing it?

**Reply by tzachi-dar** [reply]
Young people sometimes have a nanny that knows some things, but sometimes does not know how to operate the phone.

**Comment by Navid200** [comment]
@tzachi-dar   Does this mean if I allow someone to follow me, they can stop my sensor?

**Reply by tzachi-dar** [reply]
It does, but you don't just let someone follow you. You let someone you trust follow you.

**Reply by Navid200** [reply]
@tzachi-dar It could be just an accident.
Shouldn't we add a switch for the master to disable this capability?

**Reply by Navid200** [reply]
If there is a switch, we can have it disabled by default.
When a parent sets this up, the can enable it.
My concern is that lots of people use xDrip and follow but have no idea this capability exists.
If the person who is being followed is an adult, they may still allow someone to follow them but not allow them to stop the sensor.

**Comment by Navid200** [comment]
The existing switch now controls remote sensor stop as well.
This is available starting with the May 21, 2021 release of xDrip.

So, if you want someone to follow you but don't want them to be able to stop your sensor, you need to disable the switch since it is set by default.

![1](https://user-images.githubusercontent.com/51497406/119231138-3e4a3c00-baed-11eb-853e-56ab65a68d83.png)
